src/EscapeAnalysis.jl

```diff
@@ -295,34 +332,30 @@ function find_escapes(ir::IRCode, nargs::Int)
                 elseif head === :(=)
                     lhs, rhs = stmt.args
                     if isa(lhs, GlobalRef)
-                        add_change!(rhs, ir, Escape(), changes)
+                        add_change!(rhs, ir, GlobalEscape(), changes)
                     end
-                elseif head === :cfunction
-                    # for :cfunction we conservatively escapes all its arguments
-                    add_changes!(stmt.args, ir, Escape(), changes)
                 elseif head === :foreigncall
                     # for foreigncall we simply escape every argument (args[6:length(args[3])])
                     # and its name (args[1])
                     # TODO: we can apply similar strategy like builtin calls
                     #       to specialize some foreigncalls
                     foreigncall_nargs = length((stmt.args[3])::SimpleVector)
-                    push!(changes, (stmt.args[1], Escape()))
-                    add_changes!(stmt.args[6:5+foreigncall_nargs], ir, Escape(), changes)
+                    push!(changes, (stmt.args[1], ThrownEscape()))
+                    add_changes!(stmt.args[6:5+foreigncall_nargs], ir, ThrownEscape(), changes)
                 elseif is_meta_expr_head(head)
                     continue
                 elseif head === :static_parameter
-                    # static_parameter reference static parameter using index
-                    continue
+                    is_effect_free && continue
+                    add_change!(SSAValue(pc), ir, ThrownEscape(), changes)
```

Comment:
why would this be a thrown-escape? it doesn't use any values

Reply:
Hmm, yeah, I couldn't remember why I added this change..

Reply:
Well, it "uses" (refers) any of static parameters ? So I guess the better reasoning here is that we're really not interested in propagating this escape information because static parameters are already allocated from the beginning ?

Reply:
I suppose it is unclear if this `SSAValue(pc)` is the "use" site though, or if it is the "def" site. But normally, an instruction would not change `SSAValue(pc)`, only `SSAValue(pc')` for `pc' < pc`

Reply:
I had some mental model of `:static_parameter` similar to `getfield` or `arrayref`, where we have to propagate the escape information to the original object (since our `EscapeLattice` doesn't encode explicit pointer information). So I wondered we have to propagate to `ThrownEscape()` to static parameters (but this is also not needed).

Well, even from that view, it doesn't make any sense to propagate it to `SSAValue(pc)`..